carina-api/src/test/java/com/qaprosoft/apitools/validation/XmlValidatorTest.java

```diff
@@ -0,0 +1,64 @@
+package com.qaprosoft.apitools.validation;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Collectors;
+
+public class XmlValidatorTest {
+
+    @Test
+    public void testValidateXmlSuccess() throws IOException {
+        String actualXmlData = Files.lines(Path.of("src/test/resources/validation/xml_file/actual.xml"))
+                .collect(Collectors.joining("\n"));
+        XmlValidator.validateXml(actualXmlData,
+                "src/test/resources/validation/xml_file/expected.xml", XmlCompareMode.STRICT);
+    }
+
+    @Test
+    public void testValidateXmlSuccess2() throws IOException {
+        String actualXmlData = Files.lines(Path.of("src/test/resources/validation/xml_file/actual1.xml"))
+                .collect(Collectors.joining("\n"));
+        XmlValidator.validateXml(actualXmlData,
+                "src/test/resources/validation/schema/schema_xml/expected1.xml", XmlCompareMode.STRICT);
+    }
+
+    @Test
+    public void testValidateXmlError() throws IOException {
+        String actualXmlData = Files.lines(Path.of("src/test/resources/validation/xml_file/actual_error.xml"))
+                .collect(Collectors.joining("\n"));
+        boolean isErrorThrown = false;
+        try {
+            XmlValidator.validateXml(actualXmlData, "src/test/resources/validation/xml_file/expected.xml",
+                    XmlCompareMode.STRICT);
+        } catch (AssertionError e) {
+            isErrorThrown = true;
+        }
+        Assert.assertTrue(isErrorThrown, "Assertion Error not thrown");
+    }
+
+    @Test
+    public void testValidateXmlNotStrictOrderSuccess() throws IOException {
+        String actualXmlData = Files.lines(Path.of("src/test/resources/validation/xml_file/actual_order.xml"))
+                .collect(Collectors.joining("\n"));
+        XmlValidator.validateXml(actualXmlData, "src/test/resources/validation/xml_file/actual.xml",
+                XmlCompareMode.NON_STRICT);
+    }
+
+    @Test
+    public void testValidateNonExtensibleError() throws IOException {
+        String expectedXmlData = Files.lines(Path.of("src/test/resources/validation/xml_file/expected_extensible.xml"))
+                .collect(Collectors.joining("\n"));
+        boolean isErrorThrown = false;
+        try {
+            XmlValidator.validateXml(expectedXmlData,
+                    "src/test/resources/validation/xml_file/actual.xml", XmlCompareMode.NON_STRICT);
+        } catch (AssertionError e) {
+            isErrorThrown = true;
+        }
+        Assert.assertTrue(isErrorThrown, "Assertion Error not thrown");
```

Comment:
please add validation of error message as well.
in this case and all similar cases